.github/actions/build-and-push-docker-image/action.yml

```diff
@@ -0,0 +1,38 @@
+name: Build and Push Docker Image
+description: Builds docker image, logs in to ECR and pushes the new image to ECR
+inputs:
+  aws_ecr:
+    description: "The ECR base url to push the image into (omit the redis-roaring image name)"
+    required: true
+  aws_region:
+    description: "AWS region"
+    default: "us-west-2"
+  oidc_role:
+    type: string
+    required: true
+runs:
+  using: "composite"
+  steps:
+    - name: Configure AWS Credentials
+      uses: aws-actions/configure-aws-credentials@v2
+      with:
+        role-to-assume: ${{ inputs.oidc_role }}
+        role-session-name: GithubActions
+        aws-region: ${{ inputs.aws_region }}
+    - name: Login to Amazon ECR
+      id: login_ecr
+      uses: aws-actions/amazon-ecr-login@v1
+    - name: Set up QEMU
+      uses: docker/setup-qemu-action@v1
+    - name: Set up Docker Buildx
+      uses: docker/setup-buildx-action@v1
+    - name: Checkout
+      uses: actions/checkout@v3
+    - name: Build, tag, and push image to Amazon ECR
+      id: build-image
+      env:
+        AWS_ECR: ${{ inputs.aws_ecr }}
+      run: |
+        docker buildx build -t $AWS_ECR/redis-roaring:sha_$GITHUB_SHA --platform=linux/arm64 --output type=docker .
```

Comment:
What environments are we building this for? Our AWS resources run Intel, but I get if we want to also build AMD for use in local envs

Reply:
Oh; I was going off the spacey example... 

Assuming we want our ECR image to be the thing we deploy to our AWS accounts, it sounds like I should get rid of the `--platform` flag?

Reply:
Yep, you might not even need buildx either

Reply:
Removed buildx and that seems to still work fine.  Good suggestion!